docker-mender-convert

```diff
@@ -8,7 +8,8 @@ MENDER_CONVERT_DIR="$(pwd)"
 mkdir -p output
 
 docker run \
-       --mount type=bind,source="$MENDER_CONVERT_DIR,target=/mender-convert" \
+       -v $MENDER_CONVERT_DIR/input:/mender-convert/input \
+       -v $MENDER_CONVERT_DIR/output:/mender-convert/output \
```

Comment:
> Simplify docker usage by moving the source code into the container image instead of bind mounting

I don't understand.. How does this simplify things? You are making the change in the `docker-mender-convert` file, which is in the repository. So the repository is definitely available when calling this. Why not just mount the whole thing? Wouldn't you expect the currently checked out version of `mender-convert` to be called?

Reply:
I think it is a workaround that @ryansch has found to work for him to avoid the U-boot build failure in case sensitive filesystems (Mac in his case). So copying instead of bind mounting.

But for us seems to create a conflict, as we actively decided to do the opposite. The reason to mount instead of copy was to have an easier development process, right?

IMO it would be okey with the copy approach, but I was not here when we decided to do otherwise, so I leave it in your hands.

Reply:
Oh, so it's because of the rest of the source. Yeah, I did not see that link.

I don't think we should change `Dockerfile`, since we don't expect users to be actively rebuilding the container. However, I think there is a middle ground here. Instead of doing the copy in the `Dockerfile`, just make sure we copy the U-Boot source to some other location inside the container, before building it. Any non-volume location should be free of the case sensitivity issues.

Reply:
@kacf This feels similar to my PR on mender-artifact.  Docker is an incredibly useful tool for the end user.  In my case I don't have a linux box handy (that's not a VM) so it's wonderful to be able to say `docker run` to be able to use your software.  Should I attempt to package `mender-convert` differently?  How should I be able to run this on macOS?

Reply:
@ryansch I think @kacf means that we don't have to change the mount, but rather clone & build from in a different directory (not inside the mender-convert bindmount).

mender-convert$ grep clone *
convert-stage-5.sh:    git clone https://github.com/mendersoftware/grub-mender-grubenv.git $grubenv_dir >> "$build_log" 2>&1
convert-stage-5.sh:    git clone git://git.savannah.gnu.org/grub.git $grub_dir >> "$build_log" 2>&1
rpi-convert-stage-5.sh:    git clone https://github.com/mendersoftware/uboot-mender.git -b $branch >> "$build_log" 2>&1

Perhaps we can (sym/hard) link the clone destination directories to a non-volume location in the docker environment? This can be done before running the container I believe, so that it will work (i.e. not link) when running without a container.

Does that make sense?

Reply:
Hmm, I'm actually not sure I understood @estenberg's comment... :-)

This is what I mean: With the existing mount, you have `/mender-convert`. The problem is that you cannot use it as is, because of the case insensitivity issues. So just copy it to `/mender-convert-local` as part of the Docker container entrypoint script, and then perform the operation there instead, and then copy the result back to `/mender-convert`. This gives you the best of both worlds: You avoid the issues, but to the user it looks like it executed inside his original folder.

Reply:
That would work but it might not be necessary to copy all of mender-convert. The case-sensitive issues seems to happen during U-Boot compilation, and currently U-Boot is cloned into a sub-directory of mender-convert.

Rather cloning U-Boot to a non-volume location inside the container should solve it, but it does require changes to the mender-convert scripts themselves, not just the Docker build/run scripts.

Reply:
I see. Given the amount of writing done during a regular invocation of mender-convert, I don't think copying the directory is a big deal though.